circuits/kimchi/src/polynomials/chacha.rs

```diff
@@ -422,20 +432,20 @@ pub fn constraint<F: FftField>(alpha0: usize) -> E<F> {
             })
             .collect();
 
-        let mut constraints: Vec<E<F>> = low_bits.iter().map(boolean).collect();
-        constraints.push(combine_nybbles(y_xor_xprime_rotated) - yprime);
-        E::combine_constraints(alpha0, constraints)
+        let mut constraints: Vec<E<F>> = low_bits.iter().map(Helper::boolean).collect();
+        constraints.push(Helper::combine_nybbles(y_xor_xprime_rotated) - yprime);
+        E::combine_constraints(&mut alphas.by_ref().take(9), constraints)
     };
 
     let index = |g: GateType| E::cell(Column::Index(g), Curr);
     use GateType::*;
     vec![
         // a += b; d ^= a; d <<<= 16 (=4*4);
-        index(ChaCha0) * line(4),
+        index(ChaCha0) * Helper::line(&mut alphas.by_ref().take(5), 4),
```

Comment:
Guessing in the follow up these will all use the same alpha powers?

Reply:
yeah, I created functions for each chacha constraint as they are seen as different gates from the outside